### PR TITLE
Unit tests for untested methods of tardis/util.py

### PR DESCRIPTION
**Pull request by karandesai-96**
This PR will be consisting a series of commits targetting to improve code coverage of **tardis/utils.py**.

This checklist maintains the methods/ classes present in **tardis/utils.py** on 05 March 2016, the day when #506 was created. The commits enclosed in brackets show where these tests were modified/ added.
- **Classes**:
- [X] MalformedError (71bc70d)
- [X] MalformedSpeciesError (71bc70d)
- [X] MalformedElementSymbolError (71bc70d)
- [x] MalformedQuantityError (71bc70d) <br><br>
- **Methods**:
- [X] int_to_roman (fe58570)
- [X] roman_to_int (03c3f95)
- [X] calculate_luminosity (dc4abe6)
- [ ] create_synpp_yaml (excluded in this PR)
- [X] intensity_black_body (dc4abe6)
- [X] savitzky_golay (9fe1a16)
- [X] species_tuple_to_string (6bed2b4)
- [X] species_string_to_tuple (6bed2b4)
- [X] parse_quantity (9231364)
- [X] element_symbol2atomic_number (9231364)
- [X] quantity_linspace (9231364)

### CODE REVIEW

**Comment by karandesai-96** [comment]
* Till this point, tests for four classes of custom errors, and a method `int_to_roman` are added in **tardis/util.py**.
* Moreover the code for method `int_to_roman` has been improved, and it is not just about PEP8 fixes. Description can be found in 51b2a8c

**Comment by karandesai-96** [comment]
* Many tests did not check whether a method raises exceptions as defined or not, so the test_methods were improved by adding test cases and using `pytest.raises` to increase coverage by checking Exceptions raised properly.
* These methods are : `test_element_symbol2atomic_number`, `test_parse_quantity`,`species_string_to_tuple`, `species_tuple_to_string`.
* The coverage of **tardis/util.py** has reached so far till around 65%.
* The code in **tardis/tests/test_util.py** is cluttered up a bit. One next commit has just cosmetic changes:
    * Uniform code style
    * PEP8 fixes
    * Rearrangement of test_methods in same order as **tardis/util.py**

**Comment by karandesai-96** [comment]
Okay, so the code has been rearranged and looks quite clean now. Further task is to add new tests for `calculate_luminosity` and `intensity_black_body` methods from **tardis/util.py**.

**Comment by karandesai-96** [comment]
I have added these four test cases and they pass. Not sure if what are the typical values in practical cases. @wkerzendorf I would need your help to know good values for test cases. :smile:

**Comment by karandesai-96** [comment]
It is very pleasing to see red lines on coveralls.io gradually turn into green ! The code coverage of **tardis/utils.py** has touched the 80% bar after the push of 9fe1a16. :smile:

**Comment by karandesai-96** [comment]
Only one method is untested now: `create_synpp_yaml`.

File: tardis/util.py [hunk 1]
```diff
@@ -52,121 +52,114 @@ def __str__(self):
...
+        raise TypeError("Expected integer, got %s" % type(int_input))
+    if not 0 < int_input < 4000:
+        raise ValueError("Argument must be between 1 and 3999")
+
+    int_roman_tuples = [(1000, 'M'), (900, 'CM'), (500, 'D'), (400, 'CD'),
+                        (100 , 'C'), (90 , 'XC'), (50 , 'L'), (40 , 'XL'),
+                        (10  , 'X'), (9  , 'IX'), (5  , 'V'), (4  , 'IV'), (1, 'I')]
+
+    result = ''
+    for iterable, int_roman_tuple in enumerate(int_roman_tuples):
```
Comment (yeganer) [inline tardis/util.py]: I don't think you need enumerate here. There must be a way to iterate and get int and str directly from the loop.
```
for int, str in int_roman_tuples:
```
Something like that but working(haven't tested only guessed that it won't that easily).

Comment (karandesai-96) [inline tardis/util.py]: Okay, I am working upon it.

**Comment by karandesai-96** [comment]
Except for `create_synpp_yaml`, all the methods in **tardis/util.py** have been tested and coverage of this file has increased from about 55% to 81%.

`create_synpp_yaml` is kept out of scope of this PR as I am not well suited to write its test at this moment.

**Comment by karandesai-96** [comment]
@wkerzendorf @unoebauer @yeganer Is this relevant or should I close ? (Made before applying to demonstrate coding capability).

**Comment by unoebauer** [comment]
As far as I see it, parts of the ulit.py are no longer used or relevant. We should go over it at some point and figure out which of the tools therein are still relevant and which not. Afterwards, we can then worry about unit tests for those. For now I wouldn't worry about it. But let's leave the PR open so we don't forget :)

**Pull request merged**